packages/dmn-js-literal-expression/src/features/literal-expression-properties/components/LiteralExpressionPropertiesEditorComponent.js

```diff
@@ -32,6 +32,21 @@ export default class LiteralExpressionPropertiesComponent extends Component {
     });
   }
 
+  componentWillMount() {
+    this._eventBus.on('elements.changed', this.onChange);
+    console.log('componentDidUpdate',);
+  }
+
+  componentWillUnmount() {
+    this._eventBus.off('elements.changed', this.onChange);
+  }
+
+  onChange = () => {
+    this.setState({
+      name: this._viewer.getDecision().variable.name

```

Comment:
How will this work in case no variable is added to a decision? I am speaking of cases like:

```xml
  <decision id="Decision_1kfrmu8" name="Decision 1">
    <literalExpression id="LiteralExpression_0lr3hs5">
      <text>a</text>
    </literalExpression>
  </decision>
```

While it cannot be created in our tool, it's a valid DMN which would have failed due to missing variable.

Reply:
In this case, should the text be treated as a variable, or should we ignore the variable and avoid updating the state?

Reply:
Ideally, I'd expect a variable to be created when we change the name or type. I see that it's probably scope creep since we don't handle cases like [above](https://github.com/bpmn-io/dmn-js/pull/893#discussion_r1771043404). So for now, I'd only expect that we don't add another place where the component may crash. Code-wise, this means that I'd not update the state if a variable is missing completely.